src/NextGen/DonationForm/Repositories/DonationFormRepository.php

```
@@ -0,0 +1,188 @@
+            $node = Amount::make('amount')
+                ->levels(...array_map('absint', $block->attributes->levels))
+                ->allowCustomAmount()
+                ->defaultValue(50)
+                ->required();
+        } elseif ($block->name === "custom-block-editor/donor-name") {
+            $node = Name::make('name')->tap(function ($group) use ($block) {
+                $group->getNodeByName('firstName')
+                    ->label($block->attributes->firstNameLabel)
+                    ->placeholder($block->attributes->firstNamePlaceholder);
+
+                $group->getNodeByName('lastName')
+                    ->label($block->attributes->lastNameLabel)
+                    ->placeholder($block->attributes->lastNamePlaceholder)
+                    ->required($block->attributes->requireLastName);
+
+                if ($block->attributes->showHonorific) {
+                    $group->getNodeByName('honorific')
+                        ->label('Title')
+                        ->options(...$block->attributes->honorifics);
+                } else {
+                    $group->remove('honorific');
+                }
+            });
+        } elseif ($block->name === "custom-block-editor/paragraph") {
+            $node = Paragraph::make(substr(md5(mt_rand()), 0, 7))
+                ->content($block->attributes->content);
+        } elseif ($block->name === "custom-block-editor/email-field") {
+            $node = Email::make('email')->emailTag('email');
+        } elseif ($block->name === "custom-block-editor/payment-gateways") {
+            $node = PaymentGateways::make('gatewayId');
+        } elseif ($block->name === "custom-block-editor/donation-summary") {
+            $node = DonationSummary::make('donation-summary');
+        } elseif ($block->name === "custom-block-editor/company-field") {
+            $node = Text::make('company');
+        } else {
+            $node = Text::make($block->clientId);
```

Nah, let's specify the custom block and throw an exception if an unknown block name is provided.

---

Ok, let's just make sure it _does_ get done at some point. I don't want to keep kicking the can down the road.